### ISSUE TITLE
Provide an up-to-date example of how to use spring-boot-admin (server and clients) within Docker containers

### ISSUE THREAD

**Issue by fastnsilver** [codecentric/spring-boot-admin#130]
While there are examples online they are all out of date with the latest release (1.3.0).

Examples I've followed:

* https://github.com/etiennepeiniau/compose-spring-boot-admin
* https://github.com/boonen/microservices-demo

It would be good to know what to set each client's properties to

I have tried... (in .yml format)

```
spring:
  profiles: docker
  boot:
    admin:
      url: http://admin:5555
```
 
where `spring.boot.admin.url` above is the service name (within a `docker-compose.yml` file) for the Docker container hosting `spring-boot-admin` server.  This gets each client recognized but the Details button disappears after a while and each client's status updates to OFFLINE.

Here's what I've seen in the `spring-boot-admin` server logs...

```
admin_1 | 2015-11-25 03:40:41.323  INFO 1 --- [           main] s.b.c.e.t.TomcatEmbeddedServletContainer : Tomcat started on port(s): 5555 (http)
admin_1 | 2015-11-25 03:41:00.444  INFO 1 --- [nio-5555-exec-1] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring FrameworkServlet 'dispatcherServlet'
admin_1 | 2015-11-25 03:41:00.474  INFO 1 --- [nio-5555-exec-1] o.s.web.servlet.DispatcherServlet        : FrameworkServlet 'dispatcherServlet': initialization started
admin_1 | 2015-11-25 03:41:00.679  INFO 1 --- [nio-5555-exec-1] o.s.web.servlet.DispatcherServlet        : FrameworkServlet 'dispatcherServlet': initialization completed in 203 ms
admin_1 | 2015-11-25 03:41:01.159 DEBUG 1 --- [nio-5555-exec-1] d.c.b.a.controller.RegistryController    : Deliver registered applications with name=null
admin_1 | 2015-11-25 03:41:02.983  WARN 1 --- [pool-2-thread-1] d.c.b.a.services.ApplicationRegistrator  : Failed to register application as Application [id=null, name=spring-boot-application, managementUrl=http://8a47dac0a2fd:5555, healthUrl=http://8a47dac0a2fd:5555/health, serviceUrl=http://8a47dac0a2fd:5555] at spring-boot-admin (http://admin:5555/api/applications): I/O error on POST request for "http://admin:5555/api/applications":admin; nested exception is java.net.UnknownHostException: admin
admin_1 | 2015-11-25 03:41:02.987  WARN 1 --- [cTaskExecutor-1] d.c.b.a.services.ApplicationRegistrator  : Failed to register application as Application [id=null, name=spring-boot-application, managementUrl=http://8a47dac0a2fd:5555, healthUrl=http://8a47dac0a2fd:5555/health, serviceUrl=http://8a47dac0a2fd:5555] at spring-boot-admin (http://admin:5555/api/applications): I/O error on POST request for "http://admin:5555/api/applications":admin; nested exception is java.net.UnknownHostException: admin
admin_1 | 2015-11-25 03:41:03.013  WARN 1 --- [pool-2-thread-1] d.c.b.a.services.ApplicationRegistrator  : Failed to register application as Application [id=null, name=spring-boot-application, managementUrl=http://8a47dac0a2fd:5555, healthUrl=http://8a47dac0a2fd:5555/health, serviceUrl=http://8a47dac0a2fd:5555] at spring-boot-admin (http://admin:5555/api/applications): I/O error on POST request for "http://admin:5555/api/applications":admin; nested exception is java.net.UnknownHostException: admin
admin_1 | 2015-11-25 03:41:03.082  WARN 1 --- [pool-2-thread-1] d.c.b.a.services.ApplicationRegistrator  : Failed to register application as Application [id=null, name=spring-boot-application, managementUrl=http://8a47dac0a2fd:5555, healthUrl=http://8a47dac0a2fd:5555/health, serviceUrl=http://8a47dac0a2fd:5555] at spring-boot-admin (http://admin:5555/api/applications): I/O error on POST request for "http://admin:5555/api/applications":admin; nested exception is java.net.UnknownHostException: admin

```

Note I'm trying to run this all on a Mac using docker-machine, docker-compose, and docker. Versions 0.5.1, 1.5.1 and 1.9.1 respectively.

If I'm doing something silly please let me know.  This is the project (branch) I'm working off for the moment:  https://github.com/fastnsilver/grivet/tree/cloudnative-pt2.

boot-admin server example config:
* https://github.com/fastnsilver/grivet/blob/cloudnative-pt2/support/boot-admin/src/main/resources/application.yml

boot-admin client example config:
* https://github.com/fastnsilver/grivet/blob/cloudnative-pt2/support/api-gateway/src/main/resources/application.yml

docker-compose
* https://github.com/fastnsilver/grivet/blob/cloudnative-pt2/docker/docker-compose.yml

Thanks,
Chris.

**Comment by fastnsilver** [comment]
Nevermind, I figured this out myself.

For others interested, I fell back to the `@EnableDiscoveryClient` based registration of clients w/ Spring Boot Admin server instance.  

All I did was update the configs above and added a `spring-cloud-starter-eureka` dependency in each of the admin and client POMs.  I use an `environment` variable to set the active Spring profile in `docker-compose.yml`.

See this commit for how I changed things up to make everything work:

https://github.com/fastnsilver/grivet/commit/63dfbdf515f012839cdb65872b062be1bf2c0cfc

**Comment by joshiste** [comment]
If you've already eureka on board why you want to have a separate registration on boot admin and not using discovery client here as well?

And if I read your logs right, the Spring Boot Admin tries to register himself at himself.
Since the admin container has no docker-link to itself named admin the hostname is missing in his `/etc/hosts` and therefore he can't resolve the hostname. 

tl;dr
===
the Admin Servers `application.yml` is wrong. Fort the docker profile you should also use 
```diff
spring:
  profiles: docker
  boot:
    admin:
-      url: http://admin:${server.port}
+      url: http://localhost:${server.port}
```

**Comment by fastnsilver** [comment]
I removed all occurrences of `spring.boot.admin.url` as you saw.  The difficulty is when you're running Docker on a Mac you're actually running in a VM (VirtualBox in my case) so `localhost` won't work, unless perhaps you cheat and add `/etc/hosts` entry to map IP from `docker-machine ip <machine_name>` to `localhost`.  Anyhow, I'm good with the discovery approach rather than with having to add spring-boot-admin-client dependency to all client POMs.  And as for not being able to look at logs, well, in my setup, I pump logs to logstash so I can see them there.

**Comment by joshiste** [comment]
So instead of  `localhost` using the loopback ip `127.0.0.1` should do it, or not?
